ark/mibi/tiling_utils_test.py

```diff
@@ -0,0 +1,550 @@
+                },
+                'imagingPreset': {
+                    'preset': 'Normal',
+                    'aperture': '2',
+                    'displayName': 'Fine',
+                    'defaults': {
+                        'timingChoice': 7
+                    }
+                },
+                'sectionId': 8201,
+                'slideId': 5931,
+                'name': 'TheFirstFOV',
+                'timingDescription': '1 ms'
+            },
+            {
+                'scanCount': 1,
+                'centerPointMicrons': {
+                    'x': 100, 'y': 100
+                },
+                'timingChoice': 7,
+                'frameSizePixels': {
+                    'width': 2048, 'height': 2048
+                },
+                'imagingPreset': {
+                    'preset': 'Normal',
+                    'aperture': '2',
+                    'displayName': 'Fine',
+                    'defaults': {
+                        'timingChoice': 7
+                    }
+                },
+                'sectionId': 8201,
+                'slideId': 5931,
+                'name': 'TheSecondFOV',
+                'timingDescription': '1 ms'
+            }
+        ],
+        'region_start_x': [0, 50],
+        'region_start_y': [100, 150],
+        'fov_num_x': [2, 4],
+        'fov_num_y': [4, 2],
+        'x_fov_size': [5, 10],
+        'y_fov_size': [10, 5],
+        'randomize': [0, 0],
+        'moly_run': 1
+    }
+
+    # set moly point
+    sample_moly_point = {
+        "scanCount": 3,
+        "centerPointMicrons": {
+            "x": 14540,
+            "y": -10830
+        },
+        "fovSizeMicrons": 200,
+        "timingChoice": 7,
+        "frameSizePixels": {
+            "width": 128,
+            "height": 128
+        },
+        "imagingPreset": {
+            "preset": "Tuning",
+            "aperture": "3",
+            "displayName": "QC - 100µm",
+            "defaults": {
+                "timingChoice": 7
+            }
+        },
+        "standardTarget": "Moly Foil",
+        "name": "MoQC",
+        "notes": None,
+        "timingDescription": "1 ms"
+    }
+
+    with tempfile.TemporaryDirectory() as temp_dir:
+        # test 1: no randomization, no additional moly point interval, moly points between runs
+        tiled_regions_base = tiling_utils.create_tiled_regions(
+            sample_tiling_params, sample_moly_point
+        )
+
+        # get the center points created
+        sorted_center_points_run = [
+            (fov['centerPointMicrons']['x'], fov['centerPointMicrons']['y'])
+            for fov in tiled_regions_base['fovs']
+        ]
+
+        # define the actual center points in the same expected order
+        actual_center_points_run = [
+            (0, 100), (0, 110), (0, 120), (0, 130), (5, 100),
+            (5, 110), (5, 120), (5, 130), (14540, -10830),
+            (50, 150), (50, 155), (60, 150), (60, 155), (70, 150),
```

Comment:
This is pretty opaque. Is there a way to programatically generate this instead?

Reply:
As in something like: `[(x, y) for x in np.arange(0, 10, 5) for y in np.arange(100, 140, 10)] + [(14540, -10830)] + ...`?

Reply:
Yes exactly. Ideally with variables instead of constants which you define at the top. We're going to have the exact same type of test in the TMA tiling script, where want to check the coordinates were set appropriately